connector/src/main/java/org/geysermc/connector/entity/ThrowableEntity.java

```diff
@@ -84,36 +72,33 @@ protected void moveAbsoluteImmediate(GeyserSession session, Vector3f position, V
             moveEntityDeltaPacket.getFlags().add(MoveEntityDeltaPacket.Flag.TELEPORTING);
         }
 
-        if (position.getX() != lastBedrockPosition.getX()) {
+        if (this.position.getX() != position.getX()) {
             moveEntityDeltaPacket.getFlags().add(MoveEntityDeltaPacket.Flag.HAS_X);
             moveEntityDeltaPacket.setX(position.getX());
         }
-        if (position.getY() != lastBedrockPosition.getY()) {
+        if (this.position.getY() != position.getY()) {
             moveEntityDeltaPacket.getFlags().add(MoveEntityDeltaPacket.Flag.HAS_Y);
             moveEntityDeltaPacket.setY(position.getY());
         }
-        if (position.getZ() != lastBedrockPosition.getZ()) {
+        if (this.position.getZ() != position.getZ()) {
             moveEntityDeltaPacket.getFlags().add(MoveEntityDeltaPacket.Flag.HAS_Z);
             moveEntityDeltaPacket.setZ(position.getZ());
         }
-        setPosition(position);
-        lastBedrockPosition = position;
+        this.position = position;
```

Comment:
Make this and the below rotation setter `setPosition` and `setRotation` in case a superclass overrides how it's applied?